EKF/optflow_fusion.cpp

```diff
@@ -0,0 +1,280 @@
+			Kfusion[9] = SK_LOS[0]*(P[9][8]*SH_LOS[0]*SH_LOS[2]*sq(SK_LOS[4]) - P[9][0]*(SH_LOS[0]*SH_LOS[4]*SK_LOS[4] - SH_LOS[2]*SH_LOS[6]*SK_LOS[4]) + P[9][3]*SH_LOS[0]*SK_LOS[4]*(2*q0*q3 - 2*q1*q2) + P[9][2]*SH_LOS[0]*SH_LOS[1]*SK_LOS[4] + P[9][1]*SH_LOS[2]*SH_LOS[5]*SK_LOS[4] - P[9][5]*SH_LOS[0]*SH_LOS[6]*SK_LOS[4] - P[9][4]*SH_LOS[0]*SK_LOS[3]*SK_LOS[4]);
+			Kfusion[10] = SK_LOS[0]*(P[10][8]*SH_LOS[0]*SH_LOS[2]*sq(SK_LOS[4]) - P[10][0]*(SH_LOS[0]*SH_LOS[4]*SK_LOS[4] - SH_LOS[2]*SH_LOS[6]*SK_LOS[4]) + P[10][3]*SH_LOS[0]*SK_LOS[4]*(2*q0*q3 - 2*q1*q2) + P[10][2]*SH_LOS[0]*SH_LOS[1]*SK_LOS[4] + P[10][1]*SH_LOS[2]*SH_LOS[5]*SK_LOS[4] - P[10][5]*SH_LOS[0]*SH_LOS[6]*SK_LOS[4] - P[10][4]*SH_LOS[0]*SK_LOS[3]*SK_LOS[4]);
+			Kfusion[11] = SK_LOS[0]*(P[11][8]*SH_LOS[0]*SH_LOS[2]*sq(SK_LOS[4]) - P[11][0]*(SH_LOS[0]*SH_LOS[4]*SK_LOS[4] - SH_LOS[2]*SH_LOS[6]*SK_LOS[4]) + P[11][3]*SH_LOS[0]*SK_LOS[4]*(2*q0*q3 - 2*q1*q2) + P[11][2]*SH_LOS[0]*SH_LOS[1]*SK_LOS[4] + P[11][1]*SH_LOS[2]*SH_LOS[5]*SK_LOS[4] - P[11][5]*SH_LOS[0]*SH_LOS[6]*SK_LOS[4] - P[11][4]*SH_LOS[0]*SK_LOS[3]*SK_LOS[4]);
+			Kfusion[12] = SK_LOS[0]*(P[12][8]*SH_LOS[0]*SH_LOS[2]*sq(SK_LOS[4]) - P[12][0]*(SH_LOS[0]*SH_LOS[4]*SK_LOS[4] - SH_LOS[2]*SH_LOS[6]*SK_LOS[4]) + P[12][3]*SH_LOS[0]*SK_LOS[4]*(2*q0*q3 - 2*q1*q2) + P[12][2]*SH_LOS[0]*SH_LOS[1]*SK_LOS[4] + P[12][1]*SH_LOS[2]*SH_LOS[5]*SK_LOS[4] - P[12][5]*SH_LOS[0]*SH_LOS[6]*SK_LOS[4] - P[12][4]*SH_LOS[0]*SK_LOS[3]*SK_LOS[4]);
+			Kfusion[13] = SK_LOS[0]*(P[13][8]*SH_LOS[0]*SH_LOS[2]*sq(SK_LOS[4]) - P[13][0]*(SH_LOS[0]*SH_LOS[4]*SK_LOS[4] - SH_LOS[2]*SH_LOS[6]*SK_LOS[4]) + P[13][3]*SH_LOS[0]*SK_LOS[4]*(2*q0*q3 - 2*q1*q2) + P[13][2]*SH_LOS[0]*SH_LOS[1]*SK_LOS[4] + P[13][1]*SH_LOS[2]*SH_LOS[5]*SK_LOS[4] - P[13][5]*SH_LOS[0]*SH_LOS[6]*SK_LOS[4] - P[13][4]*SH_LOS[0]*SK_LOS[3]*SK_LOS[4]);
+			Kfusion[14] = SK_LOS[0]*(P[14][8]*SH_LOS[0]*SH_LOS[2]*sq(SK_LOS[4]) - P[14][0]*(SH_LOS[0]*SH_LOS[4]*SK_LOS[4] - SH_LOS[2]*SH_LOS[6]*SK_LOS[4]) + P[14][3]*SH_LOS[0]*SK_LOS[4]*(2*q0*q3 - 2*q1*q2) + P[14][2]*SH_LOS[0]*SH_LOS[1]*SK_LOS[4] + P[14][1]*SH_LOS[2]*SH_LOS[5]*SK_LOS[4] - P[14][5]*SH_LOS[0]*SH_LOS[6]*SK_LOS[4] - P[14][4]*SH_LOS[0]*SK_LOS[3]*SK_LOS[4]);
+			Kfusion[15] = SK_LOS[0]*(P[15][8]*SH_LOS[0]*SH_LOS[2]*sq(SK_LOS[4]) - P[15][0]*(SH_LOS[0]*SH_LOS[4]*SK_LOS[4] - SH_LOS[2]*SH_LOS[6]*SK_LOS[4]) + P[15][3]*SH_LOS[0]*SK_LOS[4]*(2*q0*q3 - 2*q1*q2) + P[15][2]*SH_LOS[0]*SH_LOS[1]*SK_LOS[4] + P[15][1]*SH_LOS[2]*SH_LOS[5]*SK_LOS[4] - P[15][5]*SH_LOS[0]*SH_LOS[6]*SK_LOS[4] - P[15][4]*SH_LOS[0]*SK_LOS[3]*SK_LOS[4]);
+			Kfusion[16] = SK_LOS[0]*(P[16][8]*SH_LOS[0]*SH_LOS[2]*sq(SK_LOS[4]) - P[16][0]*(SH_LOS[0]*SH_LOS[4]*SK_LOS[4] - SH_LOS[2]*SH_LOS[6]*SK_LOS[4]) + P[16][3]*SH_LOS[0]*SK_LOS[4]*(2*q0*q3 - 2*q1*q2) + P[16][2]*SH_LOS[0]*SH_LOS[1]*SK_LOS[4] + P[16][1]*SH_LOS[2]*SH_LOS[5]*SK_LOS[4] - P[16][5]*SH_LOS[0]*SH_LOS[6]*SK_LOS[4] - P[16][4]*SH_LOS[0]*SK_LOS[3]*SK_LOS[4]);
```

Comment:
The Kalman gain should be set to zero for states that are not active. For example if we are not doing 3axis mag fusion, then Kfusion[16] to Kfusion[21] should be set to zero to ensure that the corresponding terms in the covariance matrix are not inadvertently modified.